src/ui/common/src/components/pages/artifact/id/hook.ts

```diff
@@ -0,0 +1,143 @@
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useLocation, useParams } from 'react-router-dom';
+
+import { BreadcrumbLink } from '../../../../components/layouts/NavBar';
+import { handleGetArtifactResultContent } from '../../../../handlers/getArtifactResultContent';
+import { handleListArtifactResults } from '../../../../handlers/listArtifactResults';
+import { ArtifactResultResponse } from '../../../../handlers/responses/artifact';
+import { DagResultResponse } from '../../../../handlers/responses/dag';
+import { ContentWithLoadingStatus } from '../../../../reducers/artifactResultContents';
+import { ArtifactResultsWithLoadingStatus } from '../../../../reducers/artifactResults';
+import { WorkflowDagResultWithLoadingStatus } from '../../../../reducers/workflowDagResults';
+import { WorkflowDagWithLoadingStatus } from '../../../../reducers/workflowDags';
+import { AppDispatch, RootState } from '../../../../stores/store';
+import { isInitial, isLoading } from '../../../../utils/shared';
+
+export type useArtifactOutputs = {
+  breadcrumbs: BreadcrumbLink[];
+  artifactId: string;
+  artifact: ArtifactResultResponse;
+  contentWithLoadingStatus: ContentWithLoadingStatus;
+};
+
+export default function useArtifact(
+  apiKey: string,
+  id: string,
+  workflowBreadcrumbs: BreadcrumbLink[],
+  workflowDagResultId: string,
+  workflowDagWithLoadingStatus: WorkflowDagWithLoadingStatus,
+  workflowDagResultWithLoadingStatus: WorkflowDagResultWithLoadingStatus,
+  showDocumentTitle: boolean,
+  title = 'Artifact Details'
+): useArtifactOutputs {
+  const dispatch: AppDispatch = useDispatch();
+  let { artifactId } = useParams();
+  const path = useLocation().pathname;
+
+  if (id) {
+    artifactId = id;
+  }
+
+  const artifactContents = useSelector(
+    (state: RootState) => state.artifactResultContentsReducer.contents
+  );
+
+  const dagResult =
```

Comment:
This ensures we try to retrieve artifact from both objects, where `dagResults` has higher priority.